models/models.js

```diff
@@ -42,4 +42,20 @@ exports.fetchComments = (article_id) => {
       }
   return rows
    })
+}
+
+exports.addComment = (comment, article_id) => {
+   return db.query(`SELECT * FROM users WHERE username = $1`, [comment.author])
+   .then(({rows}) => {
+      if (rows.length){
+         return db.query(`INSERT INTO comments (body, author, article_id) VALUES ($1, $2, $3) RETURNING *;`, [comment.body, comment.author, article_id])
+         .then(({rows}) => {
+            return rows[0];
+         })
+      }
+      else {return Promise.reject({
+         status: 400,
+         message: 'bad request'
+      })}
+   })
```

Comment:
You wouldn't need this for the username if you removed what I mentioned in a previous comment as PSQL will throw an error that the username doesn't exist

Reply:
Would this be an error 400 or a 404?

Reply:
Well, PSQL would throw an error which you would then catch in your error handler.
If the username doesn't exist I would like a 404, it is a good request just the username isn't found.

Reply:
The issue is that I seem to be getting the same error code for things I should be getting a 404 on, or rather things end up flagged as a 400 when they shouldn't be... checking for existence was the way I went around it to avoid that issue. If I add the error code into my handler, it starts treating things that should be one code as another and I'm trying to find ways to differentiate :/ I went over this with Kev yesterday briefly and I got an error he hadn't seen before, and I only resolved it by chance.